pilot/pkg/model/mesh.go

```diff
@@ -0,0 +1,785 @@
+	istioConfigFilter = "io.istio.istio-config"
+
+	// Key Istio attribute names for mapping endpoints to subsets.
+	// For details, please see
+	// https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html
+
+	// UID is the attribute name for the mesh unique, platform-specific
+	// unique identifier for the server instance of the destination service. Mesh
+	// uses the label to uqniquely identify the endpoint when it receives updates
+	// from the Controller. Example: kubernetes://my-svc-234443-5sffe.my-namespace
+	// No two instances running anywhere in the mesh can have the same value for
+	// UID.
+	UID destinationAttribute = "destination.uid"
+
+	// SERVICE represents the fully qualified __CANONOCAL__ name of the Istio
+	// service, ex: "my-svc.my-namespace.svc.cluster.local".  Mesh uses
+	// this label to identify endpoints belonging to the same service for XDS queries
+	// involving service names. The attribute value is expected to be mesh unique,
+	// meaning services having this name are semantically understood to be the same
+	// modulo version and other labels, irrespective of the platform under which
+	// they are run, the clusters or availability zones they run in.
+	SERVICE destinationAttribute = "destination.service"
+
+	// NAME represents the short part of the service name, ex: "my-svc".
+	NAME destinationAttribute = "destination.name"
+
+	// NAMESPACE represents the namespace part of the destination service,
+	// example: "my-namespace".
+	NAMESPACE destinationAttribute = "destination.namespace"
+
+	// IP represents the IP address of the server instance, example 10.0.0.104.
+	// This IP is expected to be reachable from __THIS__ pilot. No distinction
+	// is being made for directly reachable service instances versus those
+	// behind a VIP. Istio's health discovery service will ensure that this
+	// endpoint's capacity is correctly reported accordingly.
+	IP destinationAttribute = "destination.ip"
+
+	// PORT represents the recipient port on the server IP address, Example: 443
+	PORT destinationAttribute = "destination.port"
+
+	// DOMAIN represents the domain portion of the service name, excluding
+	// the name and namespace, example: svc.cluster.local
+	DOMAIN destinationAttribute = "destination.domain"
+
+	// USER represents the __CANONICAL__ Mesh user running the destination
+	// application, example: service-account-foo. The user's identity is
+	// assumed to be trusted within the entire mesh, irrespective of which
+	// platform, cluster or availability zone this instance runs in.
+	USER destinationAttribute = "destination.user"
+)
+
+var (
+	destinationAttrSet = map[string]bool{
+		UID.stringValue():       true,
+		SERVICE.stringValue():   true,
+		NAME.stringValue():      true,
+		NAMESPACE.stringValue(): true,
+		IP.stringValue():        true,
+		PORT.stringValue():      true,
+		DOMAIN.stringValue():    true,
+		USER.stringValue():      true,
+	}
+
+	canonicalIstioDomain = "svc.cluster.local"
```

Comment:
This is also kubernetes specific stuff.

Reply:
There's a parameter defined for this. Please don't hardcode "cluster.local" anywhere, that's not even constant in kubernetes.

Reply:
This can be anything really as long as it's a mesh global name and that's the one that would be used in routing policies
@rshriram, @costinm , @ZackButcher  do one of you know what you'd like it to be?

Reply:
It should not be anything. We are not asking people to use a kubernetes suffix for their non kubernetes clusters, nor are we asking people to use istio.io domain suffix for their hosts (which is counter intuitive). The routing rules do not have any fixed domain name btw.

Reply:
@rshriram - not sure if NAMESPACE is tripping you up. But intent was to follow https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html which are Istio labels that are likely to be used Istio wide irrespective of platform. If NAMESPACE is a first class Istio concept then I don't think there's an issue. But even if NAMESPACE was just needed to accommodate policies specific to K8s where namespaces exist, it's possible this is required. If this is not a label at all, I'll be happy to nuke it from here (and https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html should be updated)

Reply:
Given the volume of comments around domain, I think the vocab doc actually makes more sense if these are intended to be labels that supply the domain name associated with the DNS name. Are we to interpret the label destination.domain as:

For each DNS names associated with the endpoint we ought to create the corresponding domain names so that we could use those for matching subsets.
Ex: a.b.acme.com would result in the following labels for "destination.domain":
b.acme.com and acme.com